src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java

```diff
@@ -82,61 +81,57 @@ public ModelingSubmissionResource(SubmissionRepository submissionRepository, Res
      * starting the exercise or starting a retry.
      *
      * @param exerciseId         the id of the exercise for which to init a participation
-     * @param principal          the current user principal
      * @param modelingSubmission the modelingSubmission to create
      * @return the ResponseEntity with status 200 (OK) and the Result as its body, or with status 4xx if the request is invalid
      */
     @PostMapping("/exercises/{exerciseId}/modeling-submissions")
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<ModelingSubmission> createModelingSubmission(@PathVariable long exerciseId, Principal principal, @RequestBody ModelingSubmission modelingSubmission) {
-        log.debug("REST request to create ModelingSubmission : {}", modelingSubmission.getModel());
-        long start = System.currentTimeMillis();
+    public ResponseEntity<ModelingSubmission> createModelingSubmission(@PathVariable long exerciseId, @RequestBody ModelingSubmission modelingSubmission) {
+        log.debug("REST request to create modeling submission: {}", modelingSubmission.getModel());
         if (modelingSubmission.getId() != null) {
-            throw new BadRequestAlertException("A new modelingSubmission cannot already have an ID", ENTITY_NAME, "idexists");
+            throw new BadRequestAlertException("A new modeling submission cannot already have an ID", ENTITY_NAME, "idexists");
         }
-        ResponseEntity<ModelingSubmission> response = handleModelingSubmission(exerciseId, principal, modelingSubmission);
-        long end = System.currentTimeMillis();
-        log.info("createModelingSubmission took {}ms for exercise {} and user {}", end - start, exerciseId, principal.getName());
-        return response;
+        return handleModelingSubmission(exerciseId, modelingSubmission);
     }
 
     /**
      * PUT /courses/{courseId}/exercises/{exerciseId}/modeling-submissions : Updates an existing modelingSubmission. This function is called by the modeling editor for saving and
```

Comment:
The comment is no longer correct. We update an existing modeling template or create a new one. Also, the `createModelingSubmission()` method returns the status `200` (ok) instead of `201` (created).

This also happens in the other cases. We can adjust this in this PR or in a follow-up.

Reply:
I know it is officially supported/allowed to use a put request to create objects, but I'm not sure we should do this in Artemis. We could be even stricter and only create objects with the post request (but that's just my personal preference).

Reply:
I basically agree and just kept this so that we do not unintentionally destroy something. I suggest we tackle this in a follow-up